frontend/tests/badges/BadgesTest.php

```diff
@@ -178,4 +194,69 @@ public function testListBadges() {
         // Get all badges through API
         $this->assertTrue(in_array($newBadge, $results));
     }
+
+    public function testAssignBadgesCronjob() {
+        global $conn;
+        // Create two badge receivers:
+        // - User 1 will receive: Problem Setter badge
+        // - User 2 will receive: Problem Setter and Contest Manager badges
+        $userOne = UserFactory::createUser();
+        $userTwo = UserFactory::createUser();
+
+        $loginTwo = self::login($userTwo);
+        ProblemsFactory::createProblemWithAuthor($userOne);
+        ProblemsFactory::createProblemWithAuthor($userTwo);
+        ContestsFactory::createContest(new ContestParams(['contestDirector' => $userTwo]));
+        $expectedUserOneResults = ['problemSetter'];
+        $expectedUserTwoResults = ['contestManager', 'problemSetter'];
+        Utils::RunAssignBadges();
+        {
+            $login = self::login($userOne);
+            // Fetch badges through apiMyList
+            $userOneBadges = BadgeController::apiMyList(new Request([
+                'auth_token' => $login->auth_token,
+                'user' => $userOne,
+            ]));
+            $results = self::getBadgesFromArray($userOneBadges['badges']);
+            $this->assertEquals(
+                count(array_intersect($expectedUserOneResults, $results)),
+                count($expectedUserOneResults)
+            );
+            $this->assertFalse(in_array('contestManager', $expectedUserOneResults));
+
+            // Fetch badges through apiUserList
+            $userTwoBadges = BadgeController::apiUserList(new Request([
+                'target_username' => $userTwo->username,
+            ]));
+            $results = self::getBadgesFromArray($userTwoBadges['badges']);
+            $this->assertEquals(
+                count(array_intersect($expectedUserTwoResults, $results)),
+                count($expectedUserTwoResults)
+            );
+
+            // Now check if notifications have been created for both users
+            $userOneNotifications = NotificationController::apiMyList(new Request([
+                'auth_token' => $login->auth_token,
+                'user' => $userOne,
+            ]));
+            $results = self::getBadgesFromNotificationContents($userOneNotifications['notifications']);
+            $this->assertEquals(
+                count(array_intersect($expectedUserOneResults, $results)),
+                count($expectedUserOneResults)
+            );
+            $this->assertFalse(in_array('contestManager', $expectedUserOneResults));
+        }
+        {
+            $login = self::login($userTwo); // Need to re-login after last apicall
```

Comment:
```suggestion
            $login = self::login($userTwo);
```

no necesitas justificar el re-login si no has hecho login con ese usuario antes.